virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -52,14 +52,18 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
 
   contextMenuRequested: Subject<{xPos: number, yPos: number, items: ContextMenuItem[]}>;
   readonly windowDeregisterEmitter: Subject<MVDWindowManagement.WindowId>;
+  private applicationManager: MVDHosting.ApplicationManagerInterface;
+  private viewportManager: MVDHosting.ViewportManagerInterface;
+  private pluginManager: MVDHosting.PluginManagerInterface;
 
   constructor(
-    @Inject(MVDHosting.Tokens.ApplicationManagerToken) private applicationManager: MVDHosting.ApplicationManagerInterface,
```

Comment:
I looked at this with Robert and now we understand the problem a little more. At first we didn't understand why sample-angular-app could use Inject in i18n, but then realized it was because the Injection items were defined within the angular modules that are importable from virtual-desktop. But that here, MVDHosting is a declared namespace and not a module, so the reference to it is somehow not acceptable to the i18n compiler.

Leonty, although we were able to confirm this behavior, the solution you came up with still bothers me because it sounds like it is avoiding fixing a bigger problem that we have. Can you do another day of research to determine if this problem is caused by a bad structure in zlux-platform, our tsconfig files, or can be fixed by using a different internationalization approach (Robert stated there were 2 angular internationalization libraries we were working with?)

It may be fine to merge this code, but first I want to know what we need to do to avoid more of this in the future, or if we can fix it easily now, because the rest of the codebase is using Inject and it seems more standardized to do that.

Reply:
I tried various workarounds but they didn't work. If we want to use `@Inject()` then we need to remove the namespace and also add an import statement. AoT compiler doesn't get along with namespaces: https://github.com/angular/angular/issues/15613
Only native Angular i18n approach(with AoT-compiler) is suitable for zlux-widgets and zlux-grid so the plugins that uses zlix-widgets and zlux-grid have to be AoT-compatible. Moreover, in next version of Angular,  i18n-support is going to be significantly improved so AoT-compatibility is important for future.

Reply:
I take it the workaround provided in that issue did not help? I'm referring to https://github.com/angular/angular/issues/15613#issuecomment-391485560